core/insertion_marker_manager.ts

```diff
@@ -232,49 +221,19 @@ export class InsertionMarkerManager {
    * @returns The insertion marker that represents the given block.
    */
   private createMarkerBlock(sourceBlock: BlockSvg): BlockSvg {
-    const imType = sourceBlock.type;
-
     eventUtils.disable();
     let result: BlockSvg;
     try {
-      result = this.workspace.newBlock(imType);
-      result.setInsertionMarker(true);
-      if (sourceBlock.saveExtraState) {
-        const state = sourceBlock.saveExtraState();
-        if (state && result.loadExtraState) {
-          result.loadExtraState(state);
-        }
-      } else if (sourceBlock.mutationToDom) {
-        const oldMutationDom = sourceBlock.mutationToDom();
-        if (oldMutationDom && result.domToMutation) {
-          result.domToMutation(oldMutationDom);
-        }
-      }
-      // Copy field values from the other block.  These values may impact the
-      // rendered size of the insertion marker.  Note that we do not care about
-      // child blocks here.
-      for (let i = 0; i < sourceBlock.inputList.length; i++) {
-        const sourceInput = sourceBlock.inputList[i];
-        if (sourceInput.name === constants.COLLAPSED_INPUT_NAME) {
-          continue; // Ignore the collapsed input.
-        }
-        const resultInput = result.inputList[i];
-        if (!resultInput) {
-          throw new Error(DUPLICATE_BLOCK_ERROR.replace('%1', 'an input'));
-        }
-        for (let j = 0; j < sourceInput.fieldRow.length; j++) {
-          const sourceField = sourceInput.fieldRow[j];
-          const resultField = resultInput.fieldRow[j];
-          if (!resultField) {
-            throw new Error(DUPLICATE_BLOCK_ERROR.replace('%1', 'a field'));
-          }
-          resultField.setValue(sourceField.getValue());
-        }
-      }
+      // 1. Serialize the source block to JSON
+      const blockJson = Blockly.serialization.blocks.save(sourceBlock);
```

Comment:
When you're working in core you need to import the individual module instead of referencing the fully qualified name! So import `blocks` from `serialization/blocks.js`, and then call `blocks.save` (same for `append`)!

That should fix the build error you're getting =)

Reply:
Got it. Making the changes.

Reply:
I imported like this -`import {save, append} from '../core/serialization/blocks'`

But getting an error like this - 

![image](https://github.com/google/blockly/assets/20411128/49ab840f-9bce-4348-b8a2-25a59f31ae9e)

![image](https://github.com/google/blockly/assets/20411128/c15ba085-9698-4dcf-a151-49875dd69519)

![image](https://github.com/google/blockly/assets/20411128/42764c01-9075-4e3e-b744-e8b8db7f8b70)

However, if I am doing this, apparently the errors are also going away - 

![image](https://github.com/google/blockly/assets/20411128/95fc1bc5-e51c-4e87-9e5c-8ca69c908821)

Reply:
I have also removed these two lines

```ts
      result.setCollapsed(sourceBlock.isCollapsed());
      result.setInputsInline(sourceBlock.getInputsInline());
      
```
 As the serialization/deserialization methods handle the collapsed state and inline inputs internally and manually setting them again would be redundant.